lib/ret/login.ex

```diff
@@ -22,4 +24,33 @@ defmodule Ret.Login do
     |> unique_constraint(:identifier_hash)
     |> put_assoc(:account, account)
   end
+
+  def update_identifier_hash(%{old_email: old_email, new_email: new_email} = _) do
+    old_identifier_hash = Account.identifier_hash_for_email(old_email)
+    new_identifier_hash = Account.identifier_hash_for_email(new_email)
+    login = Repo.get_by(Login, identifier_hash: old_identifier_hash)
+    case update_identifier_hash_for_login(login, new_identifier_hash) do
+      {:error, :no_login} -> {:error, :no_account_for_old_email}
+      {:error, :new_identifier_hash_is_not_unique} -> {:error, :new_email_already_in_use}
+      {:error, _} -> {:error, :failed_to_update_login}
+      _ -> :ok
+    end
+  end
+
+  defp update_identifier_hash_for_login(nil, new_identifier_hash) do
+    {:error, :no_login}
+  end
+
+  defp update_identifier_hash_for_login(%Login{} = login, new_identifier_hash) do
+    result = login
+    |> cast(%{identifier_hash: new_identifier_hash}, [:identifier_hash])
```

Comment:
`cast/4` is for handling a full set of params. This can be simplified to:
```elixir
|> change(identifier_hash: new_identifier_has)
```

Reply:
Resolved in 7b4ec57db4bea5e0a057382c5ed463592cd93da1